src/java.desktop/unix/classes/sun/java2d/xr/XRGraphicsConfig.java

```diff
@@ -30,10 +30,12 @@
 import sun.awt.X11GraphicsDevice;
 import sun.awt.X11GraphicsEnvironment;
 import sun.awt.image.SurfaceManager;
+import sun.awt.image.SunVolatileImage;
+import sun.awt.image.VolatileSurfaceManager;
 import sun.java2d.SurfaceData;
 
 public class XRGraphicsConfig extends X11GraphicsConfig implements
-        SurfaceManager.ProxiedGraphicsConfig {
+        SurfaceManager.ProxiedGraphicsConfig, SurfaceManager.Factory {
```

Comment:
Note that we have inconsistency here:
* For X11 the SurfaceManager.Factory is added to the parent X11GraphicsConfig and to the child XRGraphicsConfig.
* For glx the SurfaceManager.Factory is added to the parent OGLGraphicsConfig and is not added to the child CGLGraphicsConfig/GLXGraphicsConfig.